enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,475 @@
+2. MCO will drain and reboot the node.
+3. When the node is up, br-ex will be created. It means the node is ready for
+   OVN Kubernetes. So the local zone OVN Kubernetes components can be started.
+   The ovnkube-node container wrapper script will also add the node annotation
+  `k8s.ovn.org/node-subnets`, according to the HostSubnet CR of this node.
+4. On the local node, OVN Kubernetes will add all SDN nodes as hybrid overlay
+   nodes to the local zone and all OVN nodes as OVN-IC remote zones. On a remote
+   OVN node, OVN Kubernetes will remove a hybrid overlay node and add an OVN-IC
+   remote zone.
+5. MCO will uncordon the node. Pods will be recreated on the node using
+   ovn-kubernetes as the default CNI plugin.
+
+The above process will be repeated for each node until all the nodes have been
+applied to the new MachineConfig and converted to OVN Kubernetes.
+
+#### Migration Cleanup
+
+Once migration is complete, the admin can trigger the migration cleanup by:
+
+```bash
+$ oc patch Network.operator.openshift.io cluster --type='merge' --patch '{"spec":{"migration": null, "defaultNetwork":{ "openshiftSDNConfig": null}}'
+```
+
+CNO will:
+- delete the openshift-sdn DaemonSets and the related resources
+  (CustomResources, ConfigMaps, etc.).
+- redeploy ovn-kubernetes in "normal" mode (no migration mode script, hybrid
+  overlay disabled, etc.).
+- remove the migration-related labels and node annotations from the nodes.
+
+#### Operation Procedure
+
+To the migration cluster CNI plugin from OpenShift SDN to OVN Kubernetes
+complete the following procedure.
+
+1. To prepare the migration, users need to update the `network.operator` CR by
+   entering the following command:
+
+   ```bash
+   $ oc patch Network.operator.openshift.io cluster --type='merge' --patch '{"spec":{"migration":{"networkType":"OVNKubernetes","isLive":true}}}"
+   ```
+
+2. Check the cluster network operator, While CNO deploys the resources of OVN
+   Kubernetes, PROGRESSING will report `True`.
+
+   ```bash
+   $ oc get clusteroperators network
+   ```
+
+   Do not continue to the next step until PROGRESSING is listed as `False`
+
+3. To begin the MTU migration, specify the migration configuration by entering
+   the following command.
+
+   ```bash
+   $ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"spec":{"networkType":"OVNKubernetes"}'
+   ```
+
+4. As the MCO updates machines in each machine config pool, it reboots each node
+   one by one. You must wait until all the nodes are updated. Check the machine
+   config pool status by entering the following command:
+
+   ```bash
+   $ oc get mcp
+   ```
+
+5. To finalize the live migration, cleanup original CNI and redeploy the target
+   CNI in 'normal' mode, enter the following command
+
+   ```bash
+   $ oc patch Network.operator.openshift.io cluster --type='merge' --patch '{"spec":{"migration": null, "defaultNetwork":{ "openshiftSDNConfig": null}}'
```

Comment:
@pliurh This made me wonder if CNO will translate from  defaultNetwork.openshiftSDNConfig and set to defaultNetwork.OVNKubernetesConfig?

Reply:
No. Since I don't see there are any common settings that need to be translated, except MTU. In the offline migration, users are allowed to configure `defaultNetwork.OVNKubernetesConfig` before the migration. But if unset, CNO will just set it with default values. We will follow the same behavior in the live migration.

Regarding the MTU, CNO will check if it's possible to use the same cluster MTU as OSDN in OVN-K before migration. If not CNO will block the migration.